### DISCUSSION TITLE
Upgrading plugins to v2 with Rector

### DISCUSSION THREAD

**Discussion by bigvictorio** [kimai/kimai#4213] [Plugins]
Hello,

I was wondering about using https://github.com/fungus75/ReadOnlyAccessBundle in the latest Kimai version (version> 2.0.22). However, it seems the maintainer stopped developing updates for this. I tried to run it on multiple older Kimai versions and the last working version was: 1.30.11 stable (prod).

After checking the release history I noticed in the 2.x.x version there was a major change in how plugins are handled. So I updated composer.json (as it says per documentation) and the first error of the plugin disappeared. However, then there was another one.
So I tried to upgrade that code to the latest PHP version with rector as by documentation https://www.kimai.org/documentation/migration-v2.html but it does not seem to work 

I installed rector in Visual Studio Code and tried to run it, but I keep getting the same error. (Composer version 2.5.8 2023-06-09 17:13:21, rector/rector version 0.17.7, PHP 8.1.12)
Also, I tried it inside the Kimai server (ubuntu 22.04, kimai version: 2.0.29, composer:  2.5.8, rector/rector: 0.17.1, PHP: 8.1.21) but just these three errors:

1st error:
`[ERROR] Undefined constant Rector\Symfony\Set\SensiolabsSetList::FRAMEWORK_EXTRA_61 ` 

After deleting from rector.php at class "**$rectorConfig->sets([**" line `        \Rector\Symfony\Set\SensiolabsSetList::FRAMEWORK_EXTRA_61,`
2nd error:  `[ERROR] Undefined constant Rector\Symfony\Set\JMSSetList::REMOVE_JMS_INJECT    ` 

After deleting from rector.php at class "**$rectorConfig->sets([**`               \Rector\Symfony\Set\JMSSetList::REMOVE_JMS_INJECT,`
3rd error:  `[ERROR] The given paths do not match any files   `

I tried updating the PHP version in my composer.json file as some people on StackOverflow suggested, I tried everything.
Maybe I'm missing something minor, or something in the rector changed that is not updated in the [documentation.](https://www.kimai.org/documentation/migration-v2.html)

I also tried this on another plugin that does not work above versions > 1.30.11 https://www.kimai.org/store/district09-personal-calendar-pref-bundle.html but got the same results, same error from my dev environment. Rector can't update it.

I also tried to download composer.json from the DemoBundle plugin and use it for the non-updated plugins but nothing.

If anybody has an idea how to fix this I would really appreciate it
Thanks

**Comment by kevinpapst** [comment]
I stopped using Rector, because the results were mostly unusable.
At least if you don't know the source code and can't fix all the errors that show up afterwards.

Rector is good when it comes to changing annotations to attributes, everything else 🤷

**Comment by bigvictorio** [comment]
So there is no "automatic" way to update plugins to the newest version?

Also, after some testing, in the kimai server, when I used older versions of the rector (I think 0.16.0) everything went "okay"— I got a green message saying something about the plugin being successfully updated. However, the plugin still was not working for the newest version.

**Reply by kevinpapst** [reply]
Nope. Besides many other reasons the new version ships a completely different frontend. ...
